src/genworld.cpp

```diff
@@ -326,3 +330,137 @@ void GenerateWorld(GenWorldMode mode, uint size_x, uint size_y, bool reset_setti
 
 	_GenerateWorld();
 }
+
+/**
+ * Helper for CircularTileSearch to found a town on or near a given tile.
+ * @param tile The tile to try founding the town upon.
+ * @param user_data The ExternalTownData to attempt to found.
+ */
+static bool TryFoundTownNearby(TileIndex tile, void *user_data)
+{
+	ExternalTownData town = *(ExternalTownData *)user_data;
+	return (Command<CMD_FOUND_TOWN>::Post(tile, TSZ_SMALL, town.is_city, _settings_game.economy.town_layout, false, 0, town.name));
+}
+
+/**
+ * Load town data from _file_to_saveload and place its contents into the world.
+ * @return True if the JSON file is valid and has north and south extents, otherwise false.
+ */
+void LoadTownData()
+{
+	using json = nlohmann::json;
+
+	/* Load the JSON file as a string initially. We'll parse it soon. */
+	size_t filesize;
+	char *text{};
+	FILE *handle = FioFOpenFile(_file_to_saveload.name, "rb", TOWN_DATA_DIR, &filesize);
+	if (handle == nullptr) return;
+
+	text = ReallocT(text, filesize);
+	size_t read = fread(text, 1, filesize, handle);
+	fclose(handle);
+
+	if (read != filesize) return;
+
+	/* Add a trailing \0 to mark the end of the string */
+	text = ReallocT(text, filesize + 1);
+	text[filesize] = '\0';
```

Comment:
Pretty sure you should be able to read directly into a std::string (resized to filesize), instead of affing with C-style memory allocation and null-termination.

Reply:
Yes, I just can't figure out how... 😕

Reply:
Figured it out. 😃